src/core/meshcop/dataset_manager.cpp

```diff
@@ -1122,11 +1130,30 @@ void PendingDatasetBase::HandleDelayTimer(Timer &aTimer)
 
 void PendingDatasetBase::HandleDelayTimer(void)
 {
+    DelayTimerTlv *delayTimer;
+
+    // if the Delay Timer value is larger than what our Timer implementation can handle, we have to compute
+    // the remainder and wait some more.
+    if ((delayTimer = static_cast<DelayTimerTlv *>(mNetwork.Get(Tlv::kDelayTimer))) != NULL)
+    {
+        uint32_t delay = delayTimer->GetDelayTimer();
+
+        if (delay > Timer::kMaxDt)
+        {
+            delayTimer->SetDelayTimer(delay - Timer::kMaxDt);
+            StartDelayTimer();
```

Comment:
Here we update the delayTimer tlv to the remaining value (compared to `kMaxDt`)
But `StartDelayTimer()` uses `Timer::StartAt` and starts from `mNetwork.GetUpdateTime()` for the given interval from the tlv.... Does `mNetwork.GetUpdateTime()` get updated from somewhere else? 

How about if we instead restart the `mDelayTimer` here with the leftover time interval..

Reply:
Good point!